src/main/webapp/app/entities/flair-bi/fullscreen/flair-bi-fullscreen.controller.js

```diff
@@ -91,13 +94,83 @@
                 return item.name === dimension;
             })
         }
+        function getStartDateRange() {
+            var date = new Date();
+            var config = vm.currentDynamicDateRangeConfig;
+            if (config.toDate) {
+                date = moment(date).startOf(config.toDate).toDate();
+                return date;
+            }
+            return null;
+        }
+        function getStartDateRangeInterval(filterTime) {
+
+            var activeFilter = vm.dynamicDateRangeConfig.filter(function (item) {
+                return item.title === vm.currentDynamicDateRangeConfig;
+            });
+
+            var config = vm.currentDynamicDateRangeConfig;
+            if (config.toDate) {
+                return null;
+            } else if (activeFilter[0].isCustom) {
+                return filterTime + ' ' + activeFilter[0].unit;
+            } else if (activeFilter[0].period.days) {
+                return activeFilter[0].period.days + ' days';
+            } else if (activeFilter[0].period.months) {
+                return activeFilter[0].period.months + ' months';
+            }
+            return null;
+        }
+        function formatDate(date) {
+            if (!date) {
+                return null;
+            }
+            return filterParametersService.dateToString(date);
+        }
+
+        function resetTimezone(date) {
+            if (!date) {
+                return null;
+            }
+            date.setMinutes(date.getMinutes() - date.getTimezoneOffset());
+            return date;
+        }
+
+        function strToDate(date) {
+            if (!date) {
+                return null;
+            }
+            return Date.parse(date) ? new Date(date) : null;
+        }
+
+        function addDateRangeFilter(date, dimension) {
+            var filterParameters = filterParametersService.getSelectedFilter();
+            var dateRangeName = filterParametersService.buildDateRangeFilterName(dimension.name);
+            delete filterParameters[dimension.name];
+            if (!filterParameters[dateRangeName]) {
+                filterParameters[dateRangeName] = [];
+            }
+            filterParameters[dateRangeName].push(date);
+            filterParameters[dateRangeName]._meta = {
+                dataType: dimension.type,
+                valueType: 'dateRangeValueType'
+            };
+            filterParametersService.save(filterParameters);
+        }
+
+        function resetTimezoneDate(startDate) {
+            return startDate;
+        }
+
+
         function addFilterInQueryDTO() {
             var filters = JSON.parse($stateParams.filters);
             var filterKey = Object.keys(filters);
             var filterParameters = filterParametersService.getSelectedFilter();
 
             filterKey.forEach(element => {
                 var dimension = element.replace(":date-range", "");
+                dimension = dimension.replace(":custom-date", "");
```

Comment:
Don't add any new prefix with the dimension name as it will lead to issue.. We are planning to remove date range prefix also